Extension/MultipathUpgrade/Model/AvailableResources.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.ViewModelCollection;
+using TaleWorlds.Core;
+
+namespace YAPO.MultipathUpgrade.Model
+{
+    public class AvailableResources
+    {
+        public AvailableResources(bool hotkeyControl)
+        {
+            int cost = 0;
+            if (!hotkeyControl)
+            {
+                int dailyCost = GetMostAccurateDailyCost();
+                cost = dailyCost * YapoSettings.Instance.DaysToPayDailyCostsBuffer;
+            }
+
+            AvailableGold = Hero.MainHero.Clan.Gold + cost;
+        }
+
+        public int AvailableGold { get; set; }
+        public Dictionary<ItemCategory, int> ItemsOfCategoryWithCount { get; set; } = new Dictionary<ItemCategory, int>();
```

Comment:
I usually put members like these at the top of a class. Not sure how @tbeswick96 does it.

Reply:
Properties go below the ctor